consensus/service/src/tx_manager.rs

```diff
@@ -351,43 +362,89 @@ impl<E: ConsensusEnclaveProxy, L: Ledger, UI: UntrustedInterfaces> TxManager<E,
         Ok(self.enclave.txs_for_peer(&encrypted_txs?, aad, peer)?)
     }
 
-    pub fn get_encrypted_tx_by_hash(&self, tx_hash: &TxHash) -> Option<WellFormedEncryptedTx> {
-        self.lock_cache()
+    /// Get the encrypted transaction corresponding to the given hash.
+    fn get_encrypted_tx_by_hash(&self, tx_hash: &TxHash) -> Option<WellFormedEncryptedTx> {
+        self.cache
             .get(tx_hash)
             .map(|entry| entry.encrypted_tx().clone())
     }
 
-    pub fn num_entries(&self) -> usize {
-        self.lock_cache().len()
-    }
-
-    fn lock_cache(&self) -> MutexGuard<HashMap<TxHash, CacheEntry>> {
-        self.cache.lock().expect("lock poisoned")
+    fn num_entries(&self) -> usize {
+        self.cache.len()
     }
 }
 
 #[cfg(test)]
-mod tests {
+mod tx_manager_tests {
     use super::*;
     use crate::validators::DefaultTxManagerUntrustedInterfaces;
     use mc_common::logger::test_with_logger;
     use mc_consensus_enclave_mock::ConsensusServiceMockEnclave;
+    use mc_ledger_db::Ledger;
     use mc_transaction_core_test_utils::{
         create_ledger, create_transaction, initialize_ledger, AccountKey,
     };
     use rand::{rngs::StdRng, SeedableRng};
 
+    #[test_with_logger]
+    // Should return Ok when a well-formed Tx is inserted.
+    fn test_insert_proposed_tx_ok(logger: Logger) {
+        let tx_context = TxContext::default();
+
+        let mut mock_untrusted = MockUntrustedInterfaces::new();
+        // Untrusted's well-formed check should be called once each time insert_propose_tx is called.
+        mock_untrusted
+            .expect_well_formed_check()
```

Comment:
This is a pretty good example of how testing with mocks ought to go: the `expect_foo` tests that the mocked dependency gets called appropriately, and we can directly set what the return value should be instead of carefully constructing an input that gives the intended behavior (i.e. creating a LedgerDb, filling it with data, creating a TxContext for a Tx that is actually well-formed, etc...)